### ISSUE TITLE
Rating

### ISSUE THREAD

**Issue by knrafto** [Cal-CS-61A-Staff/oh-queue#12]
After being helped, students should be shown the option to rate their interaction with:
- 1-5 stars (some kind of interface where you just click a star to rate)
- Optional longer comments

open question: should we require a rating? or just be annoying about it?

**Comment by pbitutsky** [comment]
I don't think this should be a feature because there is a discrepancy between how well the student was helped and how well the student thought they were helped. TAs/Tutors/AIs who don't give away as much of the answer, but more thoroughly help students on a conceptual level, would score lower ratings. 

There are numerous other problems. It reminds me of [Black Mirror](https://en.wikipedia.org/wiki/Nosedive_(Black_Mirror)).

**Comment by pbitutsky** [comment]
Maybe non-quantitative feedback? A feature that allows students to write feedback either **to** the staff member or **about** the staff member?

**Comment by klee1121** [comment]
I think that perhaps it would be better if the student can write about tips for AIs on how they can help students out better during office hours, or if they have a particular feedback they want to leave (maybe they're grateful for the AI/Tutor/TA's help, or they think the tutor/AI/TA could've helped them better), they can voice it. All optional, of course. A rating might not be necessary.

**Comment by stella-lu** [comment]
So to be clear - the general consensus is to have non-quantitative, optional feedback, which would essentially just be a text box with some prompts and the ability to see/comment on past interactions. 

Should this be anonymized? If so, (I think it should be) the feedback shouldn't be immediately sent to the staff, but instead there should be a delay - e.g. all of the week's feedback is given to the staff at the end of the week (of course, anonymity would depend on how many people they helped that week).

**Comment by klee1121** [comment]
Yes, what you said is a good idea. A delay would work. Anonymized as in the staff member the student is writing a review about is anonymous or the student is anonymous?
We could just generalize the review to "What do you like best about how a CS61A AI (i.e. anyone, not a particular person) helped you" or "How could CS61A AIs improve their assistance?"

**Comment by stella-lu** [comment]
I was originally thinking that the student would be anonymous so that they would feel comfortable giving critique. I'm not sure if having the staff members anonymous would be helpful in a situation where, for example, a student gives a staff member serious critique that might want to be addressed, but other than that, it would be fine to have the staff member anonymous as well. 

@pbitutsky do you have any thoughts on if the staff member, as well as the student, should be anonymous?
